keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,646 @@
+            num_classes-channel result.
+    """
+
+    def __init__(
+        self,
+        classes,
+        backbone,
+        model_architecture=None,
+        input_shape=None,
+        input_tensor=None,
+        return_mask=False,
+    ):
+        if backbone.name.lower() == "vgg16" or backbone.name.lower() == "vgg19":
+            # Cannot run `isinstance` check as VGG16 and VGG19 are only functions and not classes.
+            if model_architecture not in ["fcn8s", "fcn16s", "fcn32s"]:
+                raise ValueError(
+                    "Invalid argument for parameter `model_architecture`. Accepted values are ['fcn8s', 'fcn16s', 'fcn32s']"
+                )
+            else:
+                if len(input_shape) != 4:
+                    input_tensor = utils.parse_model_inputs(input_shape, input_tensor)
+                else:
+                    input_tensor = tf.keras.layers.Input(
+                        shape=(input_shape[1], input_shape[2], input_shape[3])
+                    )
+                backbone_name = backbone.name.lower()
+                output_tensor = vgg_architecture_builder(
+                    classes=classes,
+                    backbone=backbone,
+                    model_architecture=model_architecture,
+                    input_tensor=input_tensor,
+                    input_shape=input_shape,
+                )
+
+                del backbone
+
+                data_format = tf.keras.backend.image_data_format()
+                if return_mask:
+                    if data_format == "channels_last":
+                        output_tensor = tf.math.argmax(output_tensor, axis=3)
+                        output_tensor = tf.expand_dims(output_tensor, axis=3)
+                    else:
+                        output_tensor = tf.math.argmax(output_tensor, axis=1)
+                        output_tensor = tf.expand_dims(output_tensor, axis=1)
+
+                output_tensor = tf.cast(output_tensor, tf.dtypes.float32)
+
+                super().__init__(
+                    inputs={"input_tensor": input_tensor},
+                    outputs={"output_tensor": output_tensor},
+                )
+
+                self.classes = classes
+                self.backbone = backbone_name
+                self.model_architecture = model_architecture
+                self.return_mask = return_mask
+
+        elif isinstance(backbone, tf.keras.models.Model) or isinstance(
+            backbone, tf.keras.Functional
+        ):
+            if model_architecture is not None:
+                raise ValueError(
+                    """`model_architecture` cannot be set if `backbone` is not a `keras_cv.models.VGG16` or
+                    `keras_cv.models.VGG19`. Either set `backbone` to one of the accepted values or remove
+                    the `model_architecture` argument."""
+                )
+            else:
+                if len(input_shape) != 4:
```

Comment:
Unnecessary check?

Reply:
Moving the earlier check will make this redundant, yes. Will make the move